lib/seattleflu/id3c/cli/command/etl/redcap_det_fh_airs.py

```diff
@@ -0,0 +1,910 @@
+            project_id = project.id,
+            raw_coded_values = True,
+            revision = REVISION,
+            help = help_message,
+            include_incomplete = True)(function)
+
+    return function
+
+
+@command_for_each_project
+def redcap_det_fh_airs(*, db: DatabaseSession, cache: TTLCache, det: dict,
+    redcap_record_instances: List[REDCapRecord]) -> Optional[dict]:
+
+    if redcap_record_instances is None or len(redcap_record_instances) == 0:
+        LOG.warning(f"There are no record instances. Skipping record.")
+        return None
+
+    enrollments = [record for record in redcap_record_instances if record.event_name == ENROLLMENT_EVENT_NAME]
+
+    if not len(enrollments) == 1:
+        LOG.warning(f"There are {len(enrollments)} enrollment instances for record: {redcap_record_instances[0].get('subject_id')}. Skipping record.")
+        return None
+
+    enrollment = enrollments[0]
+
+    incomplete_enrollment_instruments = {
+                instrument
+                    for instrument
+                    in REQUIRED_ENROLLMENT_INSTRUMENTS
+                    if not is_complete(instrument, enrollment)
+            }
+
+    if incomplete_enrollment_instruments:
+        LOG.debug(f"The following required enrollment instruments «{incomplete_enrollment_instruments}» are not yet marked complete.")
+        return None
+
+    # If the participant's age < 18 do not continue
+    if (enrollment['scr_age'] == "" or int(enrollment['scr_age']) < 18):
+        LOG.debug("The participant age is blank or marked as under 18. Skipping record.")
+        return None
+
+    patient_entry, patient_reference = airs_build_patient(enrollment)
+    
+    if not patient_entry:
+        LOG.warning(f"Skipping record {enrollment.get('subject_id')} with insufficient information to construct patient")
+        return None
+
+    birthdate = parse_date_from_string(enrollment.get('enr_dob'))
+    if not birthdate:
+        LOG.warning(f"Record {enrollment.get('subject_id')} has an invalid or missing `enr_dob` value")
+
+    location_resource_entries = airs_build_location_info(db, cache, enrollment)
+
+    persisted_resource_entries = [patient_entry, *location_resource_entries]
+
+    for redcap_record_instance in redcap_record_instances:
+
+        event_type = None
+        collection_method = None
+
+        if redcap_record_instance.event_name == ENROLLMENT_EVENT_NAME:
+            event_type = EventType.ENROLLMENT
+
+        # TODO: I believe there's only one collection_method: swab_and_send
+        # The study documents say: 
+        #    If they report symptoms on a survey, 
+        #    they will be prompted to self-collect a nasal swab,
+        #    which will be delivered and returned via courier, 
+        #    and complete a brief symptom survey
+        # which doesn't even rhyme.
+        elif redcap_record_instance.event_name in ENCOUNTER_EVENT_NAMES:
+            if is_complete('weekly', redcap_record_instance):
```

Comment:
It wasn't obvious to me from the redcap setup, but I know you've looked at this more in depth than me - do we know that the "weekly" instrument is required to be filled out for the event? (If not, it might make sense to relax this requirement so we aren't for example, excluding events where a sample was collected but the weekly survey wasn't filled out.)

Reply:
I believe that the weekly instrument is the initial point-of-contact between participants and the study--that is, the symptom formula on the weekly page is what dictates whether a (first) kit is sent to a participant. I don't see any instances of people getting kits without a complete weekly form. That's the basis for the encounter here.